fs/walk.ts

```diff
@@ -109,11 +109,19 @@ export async function* walk(
 
       if (isSymlink) {
         if (!followSymlinks) continue;
-        path = await Deno.realPath(path);
-        // Caveat emptor: don't assume |path| is not a symlink. realpath()
-        // resolves symlinks but another process can replace the file system
-        // entity with a different type of entity before we call lstat().
-        ({ isSymlink, isDirectory } = await Deno.lstat(path));
+        try {
+          path = await Deno.realPath(path);
+          // Caveat emptor: don't assume |path| is not a symlink. realpath()
+          // resolves symlinks but another process can replace the file system
+          // entity with a different type of entity before we call lstat().
+          ({ isSymlink, isDirectory } = await Deno.lstat(path));
+        } catch (err) {
+          // Ignore broken symlinks.
+          if (err instanceof Deno.errors.NotFound) {
+            continue;
+          }
+          throw err;
```

Comment:
I thought of that. Why would someone want to try to follow a dead link if they pick `followSymlinks = true`? You should only get a file or directory back in that scenario.

Unix programs don't error when following symlinks. For example, `find`:

```sh
+ cd /tmp/link-test
+ touch a
+ mkdir b
+ touch b/c
+ ln -s b l2b
+ ln -s /this-path-does-not-exist badlink
+ find -L /tmp/link-test
/tmp/link-test
/tmp/link-test/badlink
/tmp/link-test/l2b
/tmp/link-test/l2b/c
/tmp/link-test/b
/tmp/link-test/b/c
/tmp/link-test/a
```

Having to think about handling links defeats the purpose of the option. This is compounded by https://github.com/denoland/deno_std/issues/1359, which will never return `isFile = true`; this makes  `followSymlinks` useless in its current form.

There is the added problem that `followSymlinks = false` won't even return a symlink:

```ts
for await (const entry of walk("/tmp/link-test")) {
  console.log({ path: entry.path, isSymlink: entry.isSymlink });
}
```

Notice that neither `l2b` or `badlink` are returned:

```
└> deno run --allow-read test.ts
{ path: "/tmp/link-test", isSymlink: false }
{ path: "/tmp/link-test/b", isSymlink: false }
{ path: "/tmp/link-test/b/c", isSymlink: false }
{ path: "/tmp/link-test/a", isSymlink: false }
```

This is also bad because it means there is currently no good way to handle dead symlinks with `walk()`. (I only just now noticed this second problem, so I'll open a new issue for that.)